django_forest/resources/utils/smart_field.py

```diff
@@ -24,13 +24,33 @@ def _add_smart_fields(self, item, smart_fields, resource):
         for smart_field in smart_fields:
             self._handle_get_method(smart_field, item, resource)
 
-    def handle_smart_fields(self, queryset, resource, many=False):
+    def _get_smart_fields_for_request(self, collection, params):
```

Comment:
I'm not a big fan of this method.
Should be better to use something like that:
````python
def _get_smart_fields_for_request(self, collection, params=None):
  fields = [field for field in collection['fields'] if field['is_virtual']]
  if params and 'fields' in params:
    try:
      queried_fields = params['fields'][collection['name']]
    except KeyError:
      # should be never raised
      pass
    else:
      fields = [field for field in fields if field in queried_fields]
  return fields
````

Reply:
Hey, I feel a little uncomfortable with this solution, as it seems a little less readable. I'll try and meet in the middle, but ultimately that much nesting with so many conditions/catches combined with the mutation doesn't make for clear logic in my honest opinion.

Reply:
I'm not sure about your new version.  
I'm going to investigate more about this issue to be sure on the new implem.

Reply:
Ok, happy to work with you on this. This issue is capable of halting production so we will have to run off a fork until this is merged.